src/boot/efi/devicetree.c

```diff
@@ -104,6 +106,65 @@ EFI_STATUS devicetree_install(struct devicetree_state *state, EFI_FILE *root_dir
                         MAKE_GUID_PTR(EFI_DTB_TABLE), PHYSICAL_ADDRESS_TO_POINTER(state->addr));
 }
 
+static char* devicetree_get_compatible(const void *dtb) {
+        const struct fdt_header *dt_header = dtb;
+        uint32_t *cursor = (uint32_t *) ((char *)dt_header + be32toh(dt_header->off_dt_struct));
+        char *string_block = (char *) ((char *)dt_header + be32toh(dt_header->off_dt_strings));
+        uint32_t len;
+
+        if (be32toh(dt_header->magic) != 0xd00dfeed)
```

Comment:
use UINT32_C() around the constant, since otherwise the constant is a bit weird, since the values is not expressible as a signed int, yet it is made one, and then converted back to unsigned